packages/headless/src/app/engine-configuration.ts

```diff
@@ -54,6 +55,12 @@ export interface EngineConfiguration {
    * Defaults to `prod`.
    */
   environment?: PlatformEnvironment;
+
+  /**
+   * Specifies a custom environment for Relay, allowing integrations to override the default behavior.
+   * This is useful when Relay runs in unsupported or specialized contexts that require custom handling.
+   */
+  customRelayEnvironment?: CustomEnvironment;
```

Comment:
The problem with this is the general configuration object in relation with this new option. 

Arguably it's related to analytics, so it should be nested there ? 

But its kind of a mess when you look at everything that's available to configure Headless. And that would be yet another one. Like there's already stuff about `navigatorContextProvider`, `analytics.runTimeEnvironment`, which do not perfectly overlap, but do somewhat ?

So I'm not super thrilled about it. 


 Have you tried `engine.relay.updateConfig` instead ? Maybe in `buildShopifyEngine` in `@coveo/shopify` ? Does it work ? 

TBH, I'd prefer that until next Headless major version to revisit (yet again) the `EngineConfigurationOption` interface.

Reply:
the problem with updateConfig is that relay has already been initialized, and therefore has already generated, stored and potentially used a different clientId by that point, which is not ideal

Reply:
i could get it to work, but it would rely on a lot of internal implementation details, so it would be a very leaky abstraction that could easily break in the future if the implementation of relay or headless were to evolve

Reply:
Alternative I can see:

* Make it an internal, non documented option ? 
* The `createRelay` / `getRelayInstance` itself becomes the option ?